src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,257 @@
+    public TimelineService(final SecuritySettings securitySettings,
+                           final EventTypeCache eventTypeCache,
+                           final StorageDbRepository storageDbRepository,
+                           final TimelineSync timelineSync,
+                           final NakadiSettings nakadiSettings,
+                           final TimelineDbRepository timelineDbRepository,
+                           final TopicRepositoryHolder topicRepositoryHolder,
+                           final TransactionTemplate transactionTemplate,
+                           final UUIDGenerator uuidGenerator) {
+        this.securitySettings = securitySettings;
+        this.eventTypeCache = eventTypeCache;
+        this.storageDbRepository = storageDbRepository;
+        this.timelineSync = timelineSync;
+        this.nakadiSettings = nakadiSettings;
+        this.timelineDbRepository = timelineDbRepository;
+        this.topicRepositoryHolder = topicRepositoryHolder;
+        this.transactionTemplate = transactionTemplate;
+        this.uuidGenerator = uuidGenerator;
+    }
+
+    public void createTimeline(final String eventTypeName, final String storageId, final Client client)
+            throws ForbiddenAccessException, TimelineException, TopicRepositoryException {
+        if (!client.getClientId().equals(securitySettings.getAdminClientId())) {
+            throw new ForbiddenAccessException("Request is forbidden for user " + client.getClientId());
+        }
+
+        try {
+            final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+            final Storage storage = storageDbRepository.getStorage(storageId)
+                    .orElseThrow(() -> new UnableProcessException("No storage with id: " + storageId));
+            final Timeline activeTimeline = getTimeline(eventType);
+            final TopicRepository currentTopicRepo =
+                    topicRepositoryHolder.getTopicRepository(activeTimeline.getStorage());
+            final TopicRepository nextTopicRepo = topicRepositoryHolder.getTopicRepository(storage);
+            final List<PartitionStatistics> partitionStatistics =
+                    currentTopicRepo.loadTopicStatistics(Collections.singleton(activeTimeline.getTopic()));
+
+            final Timeline nextTimeline;
+            if (activeTimeline.isFake()) {
+                nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
+            } else {
+                final String newTopic = nextTopicRepo.createTopic(partitionStatistics.size(),
+                        eventType.getOptions().getRetentionTime());
+                nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, newTopic, new Date());
+            }
+
+            switchTimelines(activeTimeline, nextTimeline);
```

Comment:
@v-stepanov actual notification is sent here
https://github.com/zalando/nakadi/blob/6b2e65b21dd46b414601f2ef14ebf832cd8b3d5d/src/main/java/org/zalando/nakadi/service/timeline/TimelineSyncImpl.java#L135
we subscribed for timeline changes here:
https://github.com/zalando/nakadi/blob/6b2e65b21dd46b414601f2ef14ebf832cd8b3d5d/src/main/java/org/zalando/nakadi/repository/db/EventTypeCache.java#L142
we notify about changes here by TimelineSync.startTimelineUpdate and TimelineSync.finishTimelineUpdate